python/tests/graph/test_import.py

```diff
@@ -0,0 +1,230 @@
+
+
+_serving_tag = "serving_tag"
+_serving_sigdef_key = 'prediction_signature'
+# The name of the input tensor
+_tensor_input_name = "input_tensor"
+# The name of the output tensor (scalar)
+_tensor_output_name = "output_tensor"
+# The name of the variable
+_tensor_var_name = "variable"
+# The size of the input tensor
+_tensor_size = 3
+
+
+def _build_checkpointed_model(session, tmp_dir):
+    """
+    Writes a model checkpoint in the given directory. The graph is assumed to be generated
+     with _build_graph_var.
+    """
+    ckpt_path_prefix = os.path.join(tmp_dir, 'model_ckpt')
+    input_tensor = tfx.get_tensor(_tensor_input_name, session.graph)
+    output_tensor = tfx.get_tensor(_tensor_output_name, session.graph)
+    w = tfx.get_tensor(_tensor_var_name, session.graph)
+    saver = tf.train.Saver(var_list=[w])
+    _ = saver.save(session, ckpt_path_prefix, global_step=2702)
+    sig_inputs = {'input_sig': tf.saved_model.utils.build_tensor_info(input_tensor)}
+    sig_outputs = {'output_sig': tf.saved_model.utils.build_tensor_info(output_tensor)}
+    serving_sigdef = tf.saved_model.signature_def_utils.build_signature_def(
+        inputs=sig_inputs, outputs=sig_outputs)
+
+    # A rather contrived way to add signature def to a meta_graph
+    meta_graph_def = tf.train.export_meta_graph()
+
+    # Find the meta_graph file (there should be only one)
+    _ckpt_meta_fpaths = glob.glob('{}/*.meta'.format(tmp_dir))
+    assert len(_ckpt_meta_fpaths) == 1, \
+        'expected only one meta graph, but got {}'.format(','.join(_ckpt_meta_fpaths))
+    ckpt_meta_fpath = _ckpt_meta_fpaths[0]
+
+    # Add signature_def to the meta_graph and serialize it
+    # This will overwrite the existing meta_graph_def file
+    meta_graph_def.signature_def[_serving_sigdef_key].CopyFrom(serving_sigdef)
+    with open(ckpt_meta_fpath, mode='wb') as fout:
+        fout.write(meta_graph_def.SerializeToString())
+
+
+def _build_saved_model(session, saved_model_dir):
+    """
+    Saves a model in a file. The graph is assumed to be generated with _build_graph_novar.
+    """
+    builder = tf.saved_model.builder.SavedModelBuilder(saved_model_dir)
+    input_tensor = tfx.get_tensor(_tensor_input_name, session.graph)
+    output_tensor = tfx.get_tensor(_tensor_output_name, session.graph)
+    sig_inputs = {'input_sig': tf.saved_model.utils.build_tensor_info(input_tensor)}
+    sig_outputs = {'output_sig': tf.saved_model.utils.build_tensor_info(output_tensor)}
+    serving_sigdef = tf.saved_model.signature_def_utils.build_signature_def(
+        inputs=sig_inputs, outputs=sig_outputs)
+
+    builder.add_meta_graph_and_variables(
+        session, [_serving_tag], signature_def_map={_serving_sigdef_key: serving_sigdef})
+    builder.save()
+
+
+@contextlib.contextmanager
+def _make_temp_directory():
+    temp_dir = tempfile.mkdtemp()
+    try:
+        yield temp_dir
+    finally:
+        shutil.rmtree(temp_dir)
+
+def _build_graph_input(gin_function):
```

Comment:
(Context: design)
I am wondering, would we call this pattern [dependency injection](https://docs.pytest.org/en/latest/fixture.html#fixtures-a-prime-example-of-dependency-injection)?